modules/apps/transfer/types/token.go

```diff
@@ -25,3 +26,19 @@ func (t Token) Validate() error {
 
 	return nil
 }
+
+// ToCoin converts a Token to an sdk.Coin.
+//
+// It takes a Token as a parameter and returns an sdk.Coin and an error. The function
+// parses the Amount field of the Token into an sdkmath.Int and creates a new
+// sdk.Coin with the IBCDenom of the Token's Denom field and the parsed Amount.
+// If the Amount cannot be parsed, an error is returned with a wrapped error message.
+func (t Token) ToCoin() (sdk.Coin, error) {
```

Comment:
It would be nice to have a couple of tests for this just so we have it covered and so that any changes in the future to this function would cover any issues.

Reply:
@gjermundgaraba I just add unit test.
But I think should more some case and check in this func: 
- Amount is zero
- Amount is negative

Reply:
@DimitrisJim I got it. Thank you!